src/android/java/io/jxcore/node/BtToRequestSocket.java

```diff
@@ -170,77 +170,92 @@ public int GetLocalHostPort() {
     }
```

Comment:
@DrJukka Can you please upgrade this file along the lines that we discussed? For example, in run() you have this enormous if statement if srvSocket != null and then a tiny statement at the bottom if it is null. In truth that error case should be right after line 82 inside the catch statement followed by a return to exit.